apps/guardian-ui/src/admin/FederationAdmin.tsx

```diff
@@ -67,21 +67,19 @@ export const FederationAdmin: React.FC = () => {
     }
   }, [config, api]);
 
-  useEffect(() => {
-    const online = status?.federation ? status.federation.peers_online + 1 : 1;
-    const offline = status?.federation ? status.federation.peers_offline : 0;
+  const online = status?.federation ? status.federation.peers_online + 1 : 1;
```

Comment:
React flags [this commit](https://github.com/fedimint/ui/pull/144/commits/45a6e92c7881fd4d8f272f12ea20257252032c0d) as causing multiple (infinite) rerenders

Reply:
Probably need to calculate the pill props under a memo, removing `gaurdians` and `statusColor` from state

```ts
const [guardians, statusColor] = useMemo(() => {
    const online = status?.federation ? status.federation.peers_online + 1 : 1;
    const offline = status?.federation ? status.federation.peers_offline : 0;
    const totalPeers = online + offline;

    const onlinePercentage = online / totalPeers;
    const guardans = `${online} / ${totalPeers}`;

    const statusColor: 'green' | 'yellow' | 'red' =
      onlinePercentage === 1
        ? 'green'
        : onlinePercentage >= 2 / 3
        ? 'yellow'
        : 'red';

    return [guardans, statusColor];
  }, [status]);
  ```
  
  `"guardians"` string could use a better variable name?

Reply:
I appreciate the feedback and patience. I have implemented the useMemo solution and did some reading on the topic.

Reply:
@okjodom Where did React tell you about the infinite re-render? So I can pay better attention to that moving forward.

Reply:
Saw it in uncaught runtime errors
![image](https://github.com/fedimint/ui/assets/11217077/2ee927ad-67c1-42db-aaa3-73686f60851d)